service/resource/s3objectv2/resource.go

```diff
@@ -97,45 +99,38 @@ func (r *Resource) ApplyDeleteChange(ctx context.Context, obj, deleteChange inte
 }
 
 func (r *Resource) NewDeletePatch(ctx context.Context, obj, currentState, desiredState interface{}) (*framework.Patch, error) {
-	return &framework.Patch{}, nil
+	return framework.NewPatch(), nil
 }
 
-func toBucketObjectState(v interface{}) (BucketObjectState, error) {
+func toBucketObjectState(v interface{}) (map[string]BucketObjectState, error) {
 	if v == nil {
-		return BucketObjectState{}, nil
+		return map[string]BucketObjectState{}, nil
 	}
 
-	bucketObject, ok := v.(BucketObjectState)
+	bucketObjectState, ok := v.(map[string]BucketObjectState)
 	if !ok {
-		return BucketObjectState{}, microerror.Maskf(wrongTypeError, "expected '%T', got '%T'", bucketObject, v)
+		return map[string]BucketObjectState{}, microerror.Maskf(wrongTypeError, "expected '%T', got '%T'", bucketObjectState, v)
 	}
 
-	return bucketObject, nil
-
+	return bucketObjectState, nil
 }
 
 func toPutObjectInput(v interface{}) (s3.PutObjectInput, error) {
 	if v == nil {
 		return s3.PutObjectInput{}, nil
 	}
 
-	putObjectInput, ok := v.(s3.PutObjectInput)
+	bucketObject, ok := v.(BucketObjectState)
 	if !ok {
-		return s3.PutObjectInput{}, microerror.Maskf(wrongTypeError, "expected '%T', got '%T'", putObjectInput, v)
+		return s3.PutObjectInput{}, microerror.Maskf(wrongTypeError, "expected '%T', got '%T'", bucketObject, v)
 	}
 
-	return putObjectInput, nil
-}
-
-func toDeleteObjectInput(v interface{}) (s3.DeleteObjectInput, error) {
-	if v == nil {
-		return s3.DeleteObjectInput{}, nil
+	putObjectInput := s3.PutObjectInput{
+		Key:           aws.String(bucketObject.Key),
+		Body:          strings.NewReader(bucketObject.Body),
+		Bucket:        aws.String(bucketObject.Bucket),
+		ContentLength: aws.Int64(int64(len(bucketObject.Body))),
 	}
 
-	deleteObjectInput, ok := v.(s3.DeleteObjectInput)
-	if !ok {
-		return s3.DeleteObjectInput{}, microerror.Maskf(wrongTypeError, "expected '%T', got '%T'", deleteObjectInput, v)
-	}
-
-	return deleteObjectInput, nil
```

Comment:
I removed the delete object input. We delete the whole bucket instead of deleting objects.